adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/StructuredToTextTransformer.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.sink;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.format.UnexpectedFormatException;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.hydrator.common.RecordConverter;
+import com.google.common.base.Joiner;
+import org.apache.hadoop.io.Text;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Structured Record to Text converter
+ */
+public class StructuredToTextTransformer extends RecordConverter<StructuredRecord, Text> {
+  private final Joiner joiner;
+
+  public StructuredToTextTransformer(String delimiter) {
+    this.joiner = Joiner.on(delimiter);
+  }
+
+  @Override
+  public Text transform(StructuredRecord structuredRecord, Schema schema) throws IOException {
+    Schema structuredRecordSchema = structuredRecord.getSchema();
+    List<String> fields = new ArrayList<>();
+    for (Schema.Field field : structuredRecord.getSchema().getFields()) {
+      String fieldName = field.getName();
+      Schema.Field schemaField = structuredRecordSchema.getField(fieldName);
+      if (schemaField == null) {
+        throw new IllegalArgumentException("Input record does not contain the " + fieldName + " field.");
+      }
+      fields.add(convertField(structuredRecord.get(fieldName), schemaField.getSchema()));
+    }
+    return new Text(joiner.join(fields));
+  }
+
+  protected String convertField(Object field, Schema fieldSchema) throws IOException {
+    Schema.Type fieldType = fieldSchema.getType();
```

Comment:
this will fail for nullables. You need to do something like:

```
if (field == null) {
  return "";
}
Schema.Type fieldType = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
```

Reply:
please add unit tests for this class.